app/services/account_builder.rb

```diff
@@ -0,0 +1,93 @@
+    account = build_affiliate(account, params)
+    account = build_created_by(account, current_user)
+    account = build_facility(account, facility)
+    account = build_subclass(account_type, account, params)
+    account
+  end
+
+  private
+
+  # Dynamically calls a subclass method if one exists.  This is much cleaner
+  # than using `Module#prepend` with `super if defined?(super)`. Returns a
+  # modified account object.
+  def build_subclass(account_type, account, params)
+    method = "build_#{Account.account_type_to_param(account_type)}"
+    return account unless respond_to?(method)
+    account = send(method, account, params)
+  end
+
+  # Build an owner account_user.
+  # Returns the modified account object.
+  def build_account_users(account, current_user, params)
+    owner = User.find(params[:owner_user_id])
+    account.account_users.build(
+      user_id: owner.id,
+      user_role: 'Owner',
+      created_by: current_user.id,
+    )
+    account
+  end
+
+  # Set the affiliate for an account only if affiliates are supported for the
+  # given account type.  Returns the modified account object.
+  def build_affiliate(account, params)
+    return account unless account.class.included_modules.include?(AffiliateAccount)
+    account.affiliate_other = nil if params[:affiliate_id] != Affiliate.OTHER.id.to_s
+    account
+  end
+
+  # Set the created_by attribute for an account.
+  # Returns the modified account object.
+  def build_created_by(account, current_user)
+    account.created_by = current_user.id
+    account
+  end
+
+  # Set the faciltiy for an account.
+  # Returns the modified account object.
+  def build_facility(account, facility)
+    account.facility_id = facility.id if facility.present?
+    account
+  end
+
+  # Dynamically called by `build_subclass` if the account_type is NufsAccount.
+  def build_nufs_account(account, params)
+    account = build_nufs_account_expires_at(account, params)
+    account
+  end
+
+  # TODO: figure out how to refactor existing expires_at code. See both
+  # `configure_new_account` and `set_expires_at!`. Those methods should get
+  # deprecated.
+  def build_nufs_account_expires_at(account, params)
+    account.set_expires_at!
+    account
+  end
+
+  # Given an account type, create a new account and assign params.
+  # This method was only created in case we ever need to override it.
+  def new_account(account_type, params)
+    key = Account.account_type_to_param(account_type)
+    account_type.constantize.new(params[key])
+  end
+
+  # Returns a valid subclassed Account object; unless account_type is invalid,
+  # then raises an AccountController::RoutingErro exception.
+  def valid_account_type!(account_type, facility)
+    account_types = Account.creatable_account_types_for_facility(facility).map(&:to_s)
+    return account_type if account_types.include?(account_type)
+    raise ActionController::RoutingError, "invalid account_type: #{account_type}"
+  end
+
+end
```

Comment:
I originally started off having engines `prepend` this class and then call `super`, but this approach seemed a lot cleaner and solved allowing engines to extend the account builder (rather than adding a lot of extension bloat to controllers).